lib/web_ui/lib/src/engine/text_editing/text_editing.dart

```diff
@@ -48,6 +48,182 @@ void _setStaticStyleAttributes(html.HtmlElement domElement) {
   }
 }
 
+/// These style attributes are constant throughout the life time of an input
+/// element.
+///
+/// They are assigned once during the creation of the DOM element.
+void _hideAutofillElements(html.HtmlElement domElement) {
+  final html.CssStyleDeclaration elementStyle = domElement.style;
+  elementStyle
+    ..whiteSpace = 'pre-wrap'
+    ..alignContent = 'center'
+    ..padding = '0'
+    ..opacity = '1'
+    ..color = 'transparent'
+    ..backgroundColor = 'transparent'
+    ..background = 'transparent'
+    ..outline = 'none'
+    ..border = 'none'
+    ..resize = 'none'
+    ..textShadow = 'transparent'
+    ..transformOrigin = '0 0 0';
+
+  /// This property makes the input's blinking cursor transparent.
+  elementStyle.setProperty('caret-color', 'transparent');
+}
+
+/// Form that contains all the fields in the same AutofillGroup.
+///
+/// These values are to be used when autofill is enabled and there is a group of
+/// text fields with more than one text field.
+class AutofillGroup {
+  AutofillGroup(
+      {this.formElement, this.elements, this.items, this.singleElement});
+
+  final html.FormElement formElement;
+
+  final Map<String, html.HtmlElement> elements;
+
+  final Map<String, Autofill> items;
+
+  final bool singleElement;
+
+  factory AutofillGroup.fromFrameworkMessage(
+    Map<String, dynamic> focusedElementAutofill,
+    List<dynamic> fields,
+  ) {
+    final singleElement = (fields == null);
```

Comment:
Suggestion: when there are no fields can we return null instead of creating an empty autofill group?

Reply:
Why return null? AutofillGroup is the owner of the form. Even if the fields are empty this means form should have only one element.

Shall I rename the AutofillGroup? :) (Any name suggestions, I can say `AutofillForm`. Or following @yjbanov 's suggestion `EngineAutofillForm`)

Reply:
Ah ok that makes sense. I forgot about the form.

Ok, in this case, I still think we don't need a `singleElement` argument in the constructor. It can simply be a getter:
```dart
bool get isSingleElement => items.isEmpty;
// or
bool get isSingleElement => elements.isEmpty;
```

Reply:
Thanks for the suggestion. I removed the object property. We also don't need a getter. So far we only needed to use it in the factory constructor.